flowmanager/models.py

```diff
@@ -260,10 +269,17 @@ def check_flow_status(self, flow_id):
                 flow_id=flow_id, status=STATE_FAILED).first()
             if ret is not None:
                 return STATE_FAILED
-            ret = session.query(Pipelines).filter_by(
+
+            ret_pending = session.query(Pipelines).filter_by(
                 flow_id=flow_id, status=STATE_PENDING).first()
-            if ret is not None:
+            ret_success = session.query(Pipelines).filter_by(
+                flow_id=flow_id, status=STATE_SUCCESS).first()
+            # If not a single success and fail - still queued / pending
+            if (ret_pending is not None) and (ret_success is None):
                 return STATE_PENDING
+            # If no fail but at least one success - running / in progress
+            if (ret_pending is not None) and (ret_success is not None):
```

Comment:
this logic seems wrong - running should be when there's one running pipeline

Reply:
also comment needs to be updated.

Reply:
@akariv What if there is no running, pipeline still pending. Eg flow comes in the system for the first time with 3 pipelines:
```
pipeline/1 - pending
pipeline/2 - pending
pipeline/3 - pending
```
dpp hits update for pipeline/1:
```
pipeline/1 - running
pipeline/2 - pending
pipeline/3 - pending
```
Hits again with success for the same pipeline:
```
pipeline/1 - success
pipeline/2 - pending
pipeline/3 - pending
```
What is the status of the flow now?

Reply:
I would say the status should be 'running' in this case.

Reply:
@akariv great, right now happening that way. So merging